_data-prepper/common-use-cases/s3-logs.md

```diff
@@ -0,0 +1,176 @@
+2. S3 creates an S3 event notification in the SQS queue.
+3. Data Prepper polls Amazon SQS for messages and then receives a message.
+4. Data Prepper downloads the content from the S3 object.
+5. Data Prepper sends a document to OpenSearch for the content in the S3 object.
+
+
+## Pipeline overview
+
+Data Prepper supports reading data from S3 using the [`s3` source]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sources/s3/).
+
+The following diagram shows a conceptual outline of a Data Prepper pipeline reading from S3.
+
+<img src="{{site.url}}{{site.baseurl}}/images/data-prepper/s3-source/s3-pipeline.jpg" alt="S3 source architecture">{: .img-fluid}
+
+## Prerequisites
+
+Before Data Prepper can read log data from S3, you need the following prerequisites: 
+
+- An S3 bucket.
+- A log producer that writes logs to S3. The exact log producer will vary depending on your specific use case, but could include writing logs to S3 or a service such as Amazon CloudWatch.
+
+
+## Getting started
+
+Use the following steps to begin loading logs from S3 with Data Prepper.
+
+1. Create an [SQS standard queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/step-create-queue.html) for your S3 event notifications. 
+2. Configure [bucket notifications](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ways-to-add-notification-config-to-bucket.html) for SQS. Use the `s3:ObjectCreated:*` event type.
+3. Grant [AWS IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) permissions to Data Prepper for accessing SQS and S3.
+4. (Recommended) Create an [SQS dead-letter queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html) (DLQ).
+5. (Recommended) Configure an SQS re-drive policy to move failed messages into the DLQ.
+
+### Setting permissions for Data Prepper
+
+To view S3 logs, Data Prepper needs access to Amazon SQS and S3.
+Use the following example to set up permissions:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "s3-access",
+            "Effect": "Allow",
+            "Action": "s3:GetObject",
+            "Resource": "arn:aws:s3:::<YOUR-BUCKET>/*"
+        },
+        {
+            "Sid": "sqs-access",
+            "Effect": "Allow",
+            "Action": [
+                "sqs:DeleteMessage",
+                "sqs:ReceiveMessage"
+            ],
+            "Resource": "arn:aws:sqs:<YOUR-REGION>:<123456789012>:<YOUR-SQS-QUEUE>"
+        },
+        {
+            "Sid": "kms-access",
+            "Effect": "Allow",
+            "Action": "kms:Decrypt",
+            "Resource": "arn:aws:kms:<YOUR-REGION>:<123456789012>:key/<YOUR-KMS-KEY>"
+        }
+    ]
+}
+```
+
+If your S3 objects or SQS queues do not use KMS, you can remove the `kms:Decrypt` permission.
+
+### SQS dead-letter queue
+
+The are two options for how to handle errors resulting from processing S3 objects.
```

Comment:
"There are two options for handling errors resulting from S3 object processing:" (note the colon).